spec/services/finance/declaration_statement_attacher_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+RSpec.describe Finance::DeclarationStatementAttacher do
+  let(:declaration) { create(:ect_participant_declaration, cpd_lead_provider: cpd_lead_provider) }
+  let(:statement) { create(:ecf_statement, output_fee: true, deadline_date: 2.months.from_now) }
+  let(:cpd_lead_provider) { statement.cpd_lead_provider }
+
+  subject { described_class.new(participant_declaration: declaration) }
+
+  describe "#call" do
+    it "creates line item" do
+      expect {
+        subject.call
+      }.to change(Finance::StatementLineItem, :count).by(1)
+    end
+
+    it "create line item with same state as declaration" do
+      subject.call
+
+      line_item = Finance::StatementLineItem.last
+      expect(line_item.state).to eql(declaration.state)
```

Comment:
I'm assuming this is only temporary and the state will move off the declaration to the line item.

Reply:
> the state will move off the declaration to the line item.

no, state will exist on both the declaration and line item. they have different purposes.

- the line item state represents the relationship between the declaration and the statement
- the declaration state represents the state of the declaration in the lifecycle (state machine) of the declaration independent of any statements

Reply:
so we could have a paid declaration that's been clawed back or a paid declaration that's been awaiting for claw back?

Reply:
what's would that be useful for?

Reply:
> so we could have a paid declaration that's been clawed back

a clawed back declaration must have been paid at some point by definition. otherwise it would have been voided. so not sure what your point is?

> what problem is that trying to solve?

if hypothetically state were moved to the line item. how does one query the current state of a declaration?
- when a provider fetch all their declarations, we must provide the current state of declarations. how do you do that at scale?
- another example is one shouldn't be able to void something is clawed back. how do you discern the current state of a declaration to prevent the invalid state change?